### ISSUE TITLE
`kitchen converge` failing - not prioritizing env vars over ~/.aws/credentials

### ISSUE THREAD

**Issue by davidcpell** [test-kitchen/kitchen-ec2#258]
Hi all,

My `kitchen converge` command is failing. `kitchen diagnose --all` reveals this error:

> /Users/david/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.3.14/lib/aws-sdk-core/shared_credentials.rb:82:in `load_profile': Profile`default' not found in /Users/david/.aws/credentials (Aws::Errors::NoSuchProfileError)

It's true that I have no `[default]` profile in my shared credentials file. However, according to the [AWS SDK](http://docs.aws.amazon.com/sdk-for-ruby/latest/DeveloperGuide/aws-ruby-sdk-getting-started.html) docs my environment variables should be prioritized over the credentials file. My environment variables are set correctly, and if I move `~/.aws` to another location temporarily, the convergence works fine.

So I guess my questions are:

1) Any idea why the shared file is being prioritized here? Is this actually an issue for the AWS SDK rather than kitchen-ec2 or is it possible that kitchen-ec2 is doing something to have the SDK check my creds file?

2) If I wanted to just use the creds file, how could I make kitchen-ec2 look for a profile other than `[default]`?

**Comment by JaBurd** [comment]
Not sure this is a kitchen thing, it might be an aws sdk thing.  I know on our servers that have roles assigned to them in aws if they have a .aws/ folder cli calls will fail.  Delete the .aws/ folder and the calls will work again as expected.

**Comment by davidcpell** [comment]
@JaBurd I did a bit of digging and think I found the offending line:

https://github.com/test-kitchen/kitchen-ec2/blob/master/lib/kitchen/driver/aws/client.rb#L59

`shared_creds = ::Aws::SharedCredentials.new(:profile_name => profile_name)`

I used `pry` at this point in the code and got this error when running that line by hand:

```
[6] pry(Kitchen::Driver::Aws::Client)> shared_creds = ::Aws::SharedCredentials.new(:profile_name => profile_name)
Aws::Errors::NoSuchProfileError: Profile `default' not found in /Users/david/.aws/credentials
from /Users/david/.rbenv/versions/2.3.0/lib/ruby/gems/2.3.0/gems/aws-sdk-core-2.3.14/lib/aws-sdk-core/shared_credentials.rb:82:in `load_profile'
```

At this point, for me the `profile_name` variable (first argument to `Kitchen::Drive::Aws::Client::get_credentials` is `nil`. I'm guessing that the `SharedCredentials` class in the core gem is therefore using `default`. So it's not that kitchen-ec2 is looking for the credentials out of order, but that this part is using `default` if `user_profile` is `nil`. Maybe it would be better to include this code that instantiates `SharedCredentials` into the branch at line 68 so that it's totally avoided for people who are using `ENV`? BTW I'd be happy to open a PR to work on this with your or whoever's feedback.
